doc/source/user_guide/sparse.rst

```diff
@@ -35,21 +36,64 @@ large, mostly NA ``DataFrame``:
 
    df = pd.DataFrame(np.random.randn(10000, 4))
    df.iloc[:9998] = np.nan
-   sdf = df.to_sparse()
+   sdf = df.astype(pd.SparseDtype("float", np.nan))
```

Comment:
Makes sense to me, though perhaps as a followup? I don't plan to put more time into sparse personally.

Reply:
this would have to be a ``DataFrame.to_sparse()`` though I think, IOW, or is ``.sparse`` allowed on *any* DataFrame? the semantics are a bit odd on this

Reply:
@TomAugspurger can you respond to this

Reply:
> I think, IOW, or is .sparse allowed on any DataFrame? the semantics are a bit odd on this

Kinda. If you just do `df.sparse` on a dataframe without all-sparse values, we raise

```python
In [6]: pd.DataFrame({"A": [1, 2]}).sparse
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-ab0fb67ed650> in <module>
----> 1 pd.DataFrame({"A": [1, 2]}).sparse

...

~/sandbox/pandas/pandas/core/arrays/sparse.py in _validate(self, data)
   2119         dtypes = data.dtypes
   2120         if not all(isinstance(t, SparseDtype) for t in dtypes):
-> 2121             raise AttributeError(self._validation_msg)
   2122
   2123     @classmethod

AttributeError: Can only use the '.sparse' accessor with Sparse data.

```

But we also allow for `pd.DataFrame.sparse.from_spmatrix`.

Reply:
`this would have to be a DataFrame.astype('sparse') `

It would be `.astype('Sparse')` which is shorthand for `.astype(SparseDtype(float64, nan))`

Reply:
ok this is all fine; is there a test for using .sparse on non-any-sparse df?